tests/v2/validation/charts/cisbenchmark_test.go

```diff
@@ -0,0 +1,339 @@
+package charts
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/rancher/rancher/tests/v2/actions/charts"
+	"github.com/rancher/shepherd/clients/rancher"
+	"github.com/rancher/shepherd/clients/rancher/catalog"
+	management "github.com/rancher/shepherd/clients/rancher/generated/management/v3"
+	v1 "github.com/rancher/shepherd/clients/rancher/v1"
+	extencharts "github.com/rancher/shepherd/extensions/charts"
+	"github.com/rancher/shepherd/extensions/clusters"
+	"github.com/rancher/shepherd/extensions/defaults"
+	"github.com/rancher/shepherd/pkg/config"
+	namegen "github.com/rancher/shepherd/pkg/namegenerator"
+	"github.com/rancher/shepherd/pkg/session"
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+type CisBenchmarkTestSuite struct {
+	suite.Suite
+	cisProfileName      string
+	client              *rancher.Client
+	session             *session.Session
+	project             *management.Project
+	chartInstallOptions *charts.InstallOptions
+}
+
+func (c *CisBenchmarkTestSuite) TearDownSuite() {
+	c.session.Cleanup()
+}
+
+func (c *CisBenchmarkTestSuite) SetupSuite() {
+	testSession := session.NewSession()
+	c.session = testSession
+
+	client, err := rancher.NewClient("", testSession)
+	require.NoError(c.T(), err)
+
+	c.client = client
+
+	cisConfig := new(CisConfig)
+	config.LoadConfig(CisConfigFileKey, cisConfig)
+	c.cisProfileName = cisConfig.ProfileName
+	require.NotEmptyf(c.T(), c.cisProfileName, "CIS profile name required")
+
+	clusterName := client.RancherConfig.ClusterName
+	require.NotEmptyf(c.T(), clusterName, "Cluster name to install is not set")
+
+	clusterID, err := clusters.GetClusterIDByName(client, clusterName)
+	require.NoError(c.T(), err)
+
+	if cisConfig.ChartVersion == "" {
+		cisConfig.ChartVersion, err = client.Catalog.GetLatestChartVersion(charts.CISBenchmarkName, catalog.RancherChartRepo)
+		require.NoError(c.T(), err)
+	}
+
+	projectConfig := &management.Project{
+		ClusterID: clusterID,
+		Name:      CISBenchmarkProjectName,
+	}
+	createdProject, err := client.Management.Project.Create(projectConfig)
+	require.NoError(c.T(), err)
+	require.Equal(c.T(), createdProject.Name, CISBenchmarkProjectName)
+	c.project = createdProject
```

Comment:
may be I'm overlooking it but I do not see c.project used anywhere other than to get the cluster ID. if thats the case, why do we want to have a project created here? instead can we add cluster object to the struct and get the clusterID from the cluster object itself?